File: java-sdk/lib/src/main/java/Reactions.java
package ;

import .utils.HTTPClient;
import .utils.HTTPRequest;
import java.net.http.HttpResponse;
import org.apache.http.NameValuePair;

public class Reactions {
	private HTTPClient _defaultClient;
	private HTTPClient _securityClient;
	private String _serverUrl;
	private String _language;
	private String _sdkVersion;
	private String _genVersion;

	public Reactions(HTTPClient defaultClient, HTTPClient securityClient, String serverUrl, String language, String sdkVersion, String genVersion) {
		this._defaultClient = defaultClient;
		this._securityClient = securityClient;
		this._serverUrl = serverUrl;
		this._language = language;
		this._sdkVersion = sdkVersion;
		this._genVersion = genVersion;
	}
	
	
    /**
     * postApiReactions - create reaction
     *
     * This endpoint allows the client to create a reaction to a specified reactable (eg, Article, Comment, or User). For examples:
     *         * "Like"ing an Article will create a new "like" Reaction from the user for that Articles
     *         * "Like"ing that Article a second time will return the previous "like"
    **/
    public .models.operations.PostApiReactionsResponse postApiReactions(.models.operations.PostApiReactionsRequest request) throws Exception {
        String baseUrl = this._serverUrl;
        String url = .utils.Utils.generateURL(baseUrl, "/api/reactions");
        
        HTTPRequest req = new HTTPRequest();
        req.setMethod("POST");
        req.setURL(url);
        
        java.util.List<NameValuePair> queryParams = .utils.Utils.getQueryParams(request.queryParams);
        if (queryParams != null) {
            for (NameValuePair queryParam : queryParams) {
                req.addQueryParam(queryParam);
            }
        }
        
        HTTPClient client = this._securityClient;
        
        HttpResponse<byte[]> httpRes = client.send(req);

        String contentType = httpRes.headers().allValues("Content-Type").get(0);

        .models.operations.PostApiReactionsResponse res = new .models.operations.PostApiReactionsResponse() {{
        }};
        res.statusCode = Long.valueOf(httpRes.statusCode());
        res.contentType = contentType;
        
        if (httpRes.statusCode() == 200) {
        }
        else if (httpRes.statusCode() == 401) {
        }

        return res;
    }
	
	
    /**
     * postApiReactionsToggle - toggle reaction
     *
     * This endpoint allows the client to toggle the user's reaction to a specified reactable (eg, Article, Comment, or User). For examples:
     *         * "Like"ing an Article will create a new "like" Reaction from the user for that Articles
     *         * "Like"ing that Article a second time will remove the "like" from the user
    **/
    public .models.operations.PostApiReactionsToggleResponse postApiReactionsToggle(.models.operations.PostApiReactionsToggleRequest request) throws Exception {
        String baseUrl = this._serverUrl;
        String url = .utils.Utils.generateURL(baseUrl, "/api/reactions/toggle");
        
        HTTPRequest req = new HTTPRequest();
        req.setMethod("POST");
        req.setURL(url);
        
        java.util.List<NameValuePair> queryParams = .utils.Utils.getQueryParams(request.queryParams);
        if (queryParams != null) {
            for (NameValuePair queryParam : queryParams) {
                req.addQueryParam(queryParam);
            }
        }
        
        HTTPClient client = this._securityClient;
        
        HttpResponse<byte[]> httpRes = client.send(req);

        String contentType = httpRes.headers().allValues("Content-Type").get(0);

        .models.operations.PostApiReactionsToggleResponse res = new .models.operations.PostApiReactionsToggleResponse() {{
        }};
        res.statusCode = Long.valueOf(httpRes.statusCode());
        res.contentType = contentType;
        
        if (httpRes.statusCode() == 200) {
        }
        else if (httpRes.statusCode() == 401) {
        }

        return res;
    }
	
}

File: python-sdk/src/foremapi/reactions.py
import requests
from foremapi.models import operations
from . import utils

class Reactions:
    _client: requests.Session
    _security_client: requests.Session
    _server_url: str
    _language: str
    _sdk_version: str
    _gen_version: str

    def __init__(self, client: requests.Session, security_client: requests.Session, server_url: str, language: str, sdk_version: str, gen_version: str) -> None:
        self._client = client
        self._security_client = security_client
        self._server_url = server_url
        self._language = language
        self._sdk_version = sdk_version
        self._gen_version = gen_version

    
    def post_api_reactions(self, request: operations.PostAPIReactionsRequest) -> operations.PostAPIReactionsResponse:
        r"""create reaction
        This endpoint allows the client to create a reaction to a specified reactable (eg, Article, Comment, or User). For examples:
                * \"Like\"ing an Article will create a new \"like\" Reaction from the user for that Articles
                * \"Like\"ing that Article a second time will return the previous \"like\"
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/api/reactions"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = self._security_client
        
        r = client.request("POST", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.PostAPIReactionsResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            pass
        elif r.status_code == 401:
            pass

        return res

    
    def post_api_reactions_toggle(self, request: operations.PostAPIReactionsToggleRequest) -> operations.PostAPIReactionsToggleResponse:
        r"""toggle reaction
        This endpoint allows the client to toggle the user's reaction to a specified reactable (eg, Article, Comment, or User). For examples:
                * \"Like\"ing an Article will create a new \"like\" Reaction from the user for that Articles
                * \"Like\"ing that Article a second time will remove the \"like\" from the user
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/api/reactions/toggle"
        
        query_params = utils.get_query_params(request.query_params)
        
        client = self._security_client
        
        r = client.request("POST", url, params=query_params)
        content_type = r.headers.get("Content-Type")

        res = operations.PostAPIReactionsToggleResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            pass
        elif r.status_code == 401:
            pass

        return res

    